app/src/main/kotlin/com/intive/picover/common/platform/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.intive.picover.common.platform.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Error
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Loading
+import com.intive.picover.common.platform.viewmodel.LatestFirebaseStorageState.Success
+import com.intive.picover.images.repository.ImagesRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<LatestFirebaseStorageState>(Loading)
+	val uiState: StateFlow<LatestFirebaseStorageState> = _uiState
+
+	init {
+		viewModelScope.launch {
+			getImages()
+		}
+	}
+
+	private suspend fun getImages() {
+		_uiState.value = Loading
+		runCatching {
+			_uiState.value = Success(repository.fetchImages().sorted())
```

Comment:
To discuss: Do we want to skip the domain layer at this point to simplify the project (and use repositories in VM)? If not, then the sorting should be done on the domain layer.